gpApi/gpFetch.js

```diff
@@ -17,7 +19,7 @@ async function gpFetch(endpoint, data, revalidate) {
 
   let token;
   if (withAuth) {
-    token = getCookie('token');
```

Comment:
we can call the getServerCookies right here
```
token = getCookie('token');
serverToken = getServerToken()

token = serverToken || getCookie('token');
```

So you don't need to pass the token on line 4 at all.